mitiq/pec/test/test_pec_utils.py

```diff
@@ -0,0 +1,143 @@
+# Copyright (C) 2020 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Tests related to mitiq.pec.utils functions."""
+
+from pytest import mark
+import numpy as np
+from cirq import Gate, LineQubit, X, Y, Z, I, CNOT, depolarize
+from mitiq.pec.utils import (
+    _simple_pauli_deco_dict,
+    get_coefficients,
+    get_imp_sequences,
+    get_one_norm,
+    get_probabilities,
+)
+from mitiq.utils import _operation_to_choi
+
+BASE_NOISE = 0.01
+DECO_DICT = _simple_pauli_deco_dict(BASE_NOISE)
+DECO_DICT_SIMP = _simple_pauli_deco_dict(BASE_NOISE, simplify_paulis=True)
+NOISELESS_DECO_DICT = _simple_pauli_deco_dict(0)
+
+
+def test_simple_pauli_deco_dict_CNOT(gate: Gate = CNOT):
+    epsilon = BASE_NOISE * 4.0 / 3.0
+    c_neg = -(1 / 4) * epsilon / (1 - epsilon)
+    c_pos = 1 - 3 * c_neg
+    qreg = LineQubit.range(2)
+
+    deco = DECO_DICT[gate.on(*qreg)]
+    first_coefficient, first_imp_seq = deco[0]
+    assert np.isclose(c_pos * c_pos, first_coefficient)
+    assert first_imp_seq == [gate.on(*qreg)]
+    second_coefficient, second_imp_seq = deco[1]
+    assert np.isclose(c_pos * c_neg, second_coefficient)
+    assert second_imp_seq == [gate.on(*qreg), X.on(qreg[0])]
+    last_coefficient, last_imp_seq = deco[-1]
+    assert np.isclose(c_neg * c_neg, last_coefficient)
+    assert last_imp_seq == [gate.on(*qreg), Z.on(qreg[0]), Z.on(qreg[1])]
```

Comment:
There's a decent number of checks here - I think it would be useful to add a doc string to this test (and other tests below) to state what is being tested.